elyra-ai/elyra
# Add language check on snippets drag and drop into Script Editors

Fixes #1540

Part of these changes have been adapted from https://github.com/jupytercalpoly/jupyterlab-code-snippets/
Co-authored-by: Ai-Vy Dang, Jay Ahn, Kiran Pinnipati


### What changes were proposed in this pull request?
This PR adds language check when dragging and dropping code snippets into Elyra Script Editors (Python and R).

### How was this pull request tested?
Manual testing:
- Drag and drop a code snippet of a language that matches the editor's: automatic paste code into the editor
- Drag and drop a code snippet of a language that does not match the editor's: show warning dialog for user to confirm or cancel the paste action.

**Note**
This PR is at a WIP stage.
**Bug**: Original code editor drop event is being fired and code is pasted before language check confirmation. 
If user cancels the dialog, too late, the code has been already pasted. If user confirms the action, the code is pasted twice.


Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.
